### ISSUE TITLE
Crash rustc

### ISSUE THREAD

**Issue by ssrlive** [rust-lang/rust#125646]
```
thread 'rustc' panicked at compiler\rustc_codegen_ssa\src\back\link.rs:2700:27:
index out of bounds: the len is 29 but the index is 29
stack backtrace:
   0:     0x7ff8bd9a3b5d - std::backtrace_rs::backtrace::dbghelp64::trace
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\..\..\backtrace\src\backtrace\dbghelp64.rs:91
   1:     0x7ff8bd9a3b5d - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff8bd9a3b5d - std::sys_common::backtrace::_print_fmt
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\sys_common\backtrace.rs:68
   3:     0x7ff8bd9a3b5d - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\sys_common\backtrace.rs:44
   4:     0x7ff8bd9d4af9 - core::fmt::rt::Argument::fmt
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\core\src\fmt\rt.rs:165
   5:     0x7ff8bd9d4af9 - core::fmt::write
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\core\src\fmt\mod.rs:1168
   6:     0x7ff8bd99a2e1 - std::io::Write::write_fmt<std::sys::pal::windows::stdio::Stderr>
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\io\mod.rs:1835
   7:     0x7ff8bd9a3936 - std::sys_common::backtrace::print
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\sys_common\backtrace.rs:34
   8:     0x7ff8bd9a6b08 - std::panicking::default_hook::closure$1
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\panicking.rs:271
   9:     0x7ff8bd9a67b6 - std::panicking::default_hook
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\panicking.rs:298
  10:     0x7ff8b6b7688c - memchr
  11:     0x7ff8bd9a7[137](https://github.com/nulldotblack/wintun/actions/runs/9266177251/job/25489812616#step:3:138) - alloc::boxed::impl$50::call
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\alloc\src\boxed.rs:2077
  12:     0x7ff8bd9a7137 - std::panicking::rust_panic_with_hook
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\panicking.rs:799
  13:     0x7ff8bd9a6fc7 - std::panicking::begin_panic_handler::closure$0
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\panicking.rs:664
  14:     0x7ff8bd9a44cf - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\sys_common\backtrace.rs:171
  15:     0x7ff8bd9a6c78 - std::panicking::begin_panic_handler
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\panicking.rs:652
  16:     0x7ff8bd9fe344 - core::panicking::panic_fmt
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\core\src\panicking.rs:72
  17:     0x7ff8bd9fe57e - core::panicking::panic_bounds_check
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\core\src\panicking.rs:274
  18:     0x7ff8b5504e93 - rustc_codegen_ssa[b16efbcf849bbe15]::back::link::linker_and_flavor
  19:     0x7ff8b54f94b0 - rustc_codegen_ssa[b16efbcf849bbe15]::back::link::each_linked_rlib
  20:     0x7ff8b54f3ddb - rustc_codegen_ssa[b16efbcf849bbe15]::back::link::link_binary
  21:     0x7ff8b2ac2106 - <rustc_codegen_llvm[ae9deb7a2fb2044c]::LlvmCodegenBackend as rustc_codegen_ssa[b16efbcf849bbe15]::traits::backend::CodegenBackend>::link
  22:     0x7ff8b2a[145](https://github.com/nulldotblack/wintun/actions/runs/9266177251/job/25489812616#step:3:146)59 - <rustc_interface[dba02352c3517abd]::queries::Linker>::link
  23:     0x7ff8b29c3bce - llvm::function_ref<void __cdecl(llvm::Value const * __ptr64)>::callback_fn<`llvm::Value::stripInBoundsOffsets'::`1'::<lambda_1_1> >
  24:     0x7ff8b29c06ef - llvm::function_ref<void __cdecl(llvm::Value const * __ptr64)>::callback_fn<`llvm::Value::stripInBoundsOffsets'::`1'::<lambda_1_1> >
  25:     0x7ff8b29c8549 - llvm::function_ref<void __cdecl(llvm::Value const * __ptr64)>::callback_fn<`llvm::Value::stripInBoundsOffsets'::`1'::<lambda_1_1> >
  26:     0x7ff8bd9b818d - alloc::boxed::impl$48::call_once
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\alloc\src\boxed.rs:2063
  27:     0x7ff8bd9b818d - alloc::boxed::impl$48::call_once
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\alloc\src\boxed.rs:2063
  28:     0x7ff8bd9b818d - std::sys::pal::windows::thread::impl$0::new::thread_start
                               at /rustc/84b40fc908c3adc7e0e470b3fbaa264df0e122b8/library\std\src\sys\pal\windows\thread.rs:52
  29:     0x7ff8f39c4cb0 - BaseThreadInitThunk
  30:     0x7ff8f481e8ab - RtlUserThreadStart

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: please attach the file at `D:\a\wintun\wintun\rustc-ice-2024-05-28T08_39_42-4064.txt` to your bug report

note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental=[REDACTED]
```

**Comment by lqd** [comment]
Is there code available online so we can try to reproduce the issue? (But it's probably a duplicate of #125474 or of #125484.)

**Comment by ssrlive** [comment]
Not fixed yet.
```
Compiling wintun v0.4.0 (D:\a\wintun\wintun)
thread 'rustc' panicked at compiler\rustc_codegen_ssa\src\back\link.rs:2700:27:
index out of bounds: the len is 28 but the index is 28
stack backtrace:
   0:     0x7ff9c71d3b9d - std::backtrace_rs::backtrace::dbghelp64::trace
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\..\..\backtrace\src\backtrace\dbghelp64.rs:91
   1:     0x7ff9c71d3b9d - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff9c71d3b9d - std::sys_common::backtrace::_print_fmt
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\sys_common\backtrace.rs:68
   3:     0x7ff9c71d3b9d - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\sys_common\backtrace.rs:44
   4:     0x7ff9c72047d9 - core::fmt::rt::Argument::fmt
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\core\src\fmt\rt.rs:165
   5:     0x7ff9c72047d9 - core::fmt::write
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\core\src\fmt\mod.rs:1168
   6:     0x7ff9c71ca3f1 - std::io::Write::write_fmt<std::sys::pal::windows::stdio::Stderr>
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\io\mod.rs:1835
   7:     0x7ff9c71d3976 - std::sys_common::backtrace::print
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\sys_common\backtrace.rs:34
   8:     0x7ff9c71d6b58 - std::panicking::default_hook::closure$1
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\panicking.rs:271
   9:     0x7ff9c71d6806 - std::panicking::default_hook
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\panicking.rs:298
  10:     0x7ff9bbc260fc - memchr
  11:     0x7ff9c71d7187 - alloc::boxed::impl$50::call
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\alloc\src\boxed.rs:2077
  12:     0x7ff9c71d7187 - std::panicking::rust_panic_with_hook
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\panicking.rs:799
  13:     0x7ff9c71d7017 - std::panicking::begin_panic_handler::closure$0
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\panicking.rs:664
  14:     0x7ff9c71d450f - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\sys_common\backtrace.rs:171
  15:     0x7ff9c71d6cc8 - std::panicking::begin_panic_handler
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\panicking.rs:652
  16:     0x7ff9c722e054 - core::panicking::panic_fmt
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\core\src\panicking.rs:72
  17:     0x7ff9c722e28e - core::panicking::panic_bounds_check
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\core\src\panicking.rs:274
  18:     0x7ff9ba572796 - rustc_codegen_ssa[17375fc6d9732a23]::back::link::linker_and_flavor
  19:     0x7ff9ba566d70 - rustc_codegen_ssa[17375fc6d9732a23]::back::link::each_linked_rlib
  20:     0x7ff9ba56[167](https://github.com/ssrlive/wintun/actions/runs/9400447158/job/25890084673#step:3:168)e - rustc_codegen_ssa[17375fc6d9732a23]::back::link::link_binary
  21:     0x7ff9b7b34ca6 - <rustc_codegen_llvm[3d446f3b6ccd7211]::LlvmCodegenBackend as rustc_codegen_ssa[17375fc6d9732a23]::traits::backend::CodegenBackend>::link
  22:     0x7ff9b7a874e9 - <rustc_interface[b17a26a17ab0bfc1]::queries::Linker>::link
  23:     0x7ff9b7a358f2 - _wpgmptr
  24:     0x7ff9b7a3256f - _wpgmptr
  25:     0x7ff9b7a3a269 - _wpgmptr
  26:     0x7ff9c71e838d - alloc::boxed::impl$48::call_once
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\alloc\src\boxed.rs:2063
  27:     0x7ff9c71e838d - alloc::boxed::impl$48::call_once
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\alloc\src\boxed.rs:2063
  28:     0x7ff9c71e838d - std::sys::pal::windows::thread::impl$0::new::thread_start
                               at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library\std\src\sys\pal\windows\thread.rs:52
  29:     0x7ff9fc684cb0 - BaseThreadInitThunk
  30:     0x7ff9fdb7e8ab - RtlUserThreadStart
error: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md
note: please make sure that you have updated to the latest nightly
note: please attach the file at `D:\a\wintun\wintun\rustc-ice-2024-06-06T12_02_11-6840.txt` to your bug report
note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental=[REDACTED]
note: some of the compiler flags provided by cargo are hidden
query stack during panic:
end of query stack
error: could not compile `wintun` (lib test)
warning: build failed, waiting for other jobs to finish...
```

**Comment by geassed-bum** [comment]
It looks like it is still not fixed as of 1.80.0

> thread 'rustc' panicked at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9\compiler\rustc_middle\src\hir\mod.rs:188:33:
index out of bounds: the len is 11223 but the index is 2660552760
stack backtrace:
   Compiling lewton v0.10.2
   0:     0x7ffbb9453b3d - std::backtrace_rs::backtrace::dbghelp64::trace
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\..\..\backtrace\src\backtrace\dbghelp64.rs:91
   1:     0x7ffbb9453b3d - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ffbb9453b3d - std::sys_common::backtrace::_print_fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:68
   3:     0x7ffbb9453b3d - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:44
   4:     0x7ffbb9484779 - core::fmt::rt::Argument::fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\fmt\rt.rs:165
   5:     0x7ffbb9484779 - core::fmt::write
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\fmt\mod.rs:1168
   6:     0x7ffbb944a391 - std::io::Write::write_fmt<std::sys::pal::windows::stdio::Stderr>
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\io\mod.rs:1835
   7:     0x7ffbb9453916 - std::sys_common::backtrace::print
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:34
   8:     0x7ffbb9456af8 - std::panicking::default_hook::closure$1
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:271
   9:     0x7ffbb94567a6 - std::panicking::default_hook
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:298
   Compiling fnv v1.0.7
  10:     0x7ffbb550a39c - memchr
  11:     0x7ffbb9457127 - alloc::boxed::impl$50::call
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2077
  12:     0x7ffbb9457127 - std::panicking::rust_panic_with_hook
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:799
  13:     0x7ffbb9456fb7 - std::panicking::begin_panic_handler::closure$0
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:664
  14:     0x7ffbb94544af - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:171
  15:     0x7ffbb9456c68 - std::panicking::begin_panic_handler
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:652
  16:     0x7ffbb94ad124 - core::panicking::panic_fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\panicking.rs:72
  17:     0x7ffbb94ad35e - core::panicking::panic_bounds_check
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\panicking.rs:274
  18:     0x7ffbb41c7019 - <tracing_subscriber[9cea76178f68bf35]::registry::sharded::Registry as core[788d601b121ab5d5]::default::Default>::default
  19:     0x7ffbb4be185a - rustc_query_impl[f16a2bc9f261c9b3]::plumbing::query_key_hash_verify_all
  20:     0x7ffbb4b2a264 - rustc_ty_utils[8ea27dfdeb4a5d12]::ty::self_ty_of_trait_impl_enabling_order_dep_trait_object_hack
  21:     0x7ffbb4be2c43 - rustc_query_impl[f16a2bc9f261c9b3]::plumbing::query_key_hash_verify_all
  22:     0x7ffbb538441c - <rustc_middle[1c821902dd0c6a84]::hir::map::Map>::attrs
  23:     0x7ffbb49f59d0 - <pulldown_cmark[abc25a02e03f5439]::parse::Parser as core[788d601b121ab5d5]::iter::traits::iterator::Iterator>::next
  24:     0x7ffbb49e5067 - <pulldown_cmark[abc25a02e03f5439]::parse::Parser as core[788d601b121ab5d5]::iter::traits::iterator::Iterator>::next
  25:     0x7ffbb4a02295 - <rustc_passes[26ec330937c00dd5]::check_attr::CheckAttrVisitor as rustc_hir[14dff18277ee88c]::intravisit::Visitor>::visit_impl_item
  26:     0x7ffbb4a024cd - rustc_passes[26ec330937c00dd5]::check_attr::check_mod_attrs
  27:     0x7ffbb3db3efa - <dyn std[4f3d19e5cfafb1ab]::io::Write as nu_ansi_term[383177f773185f1f]::write::AnyWrite>::write_str
  28:     0x7ffbb3d43f78 - rustc_ty_utils[8ea27dfdeb4a5d12]::ty::adt_sized_constraint
   Compiling vec_map v0.8.2
  29:     0x7ffbb3dc74b1 - rustc_query_impl[f16a2bc9f261c9b3]::query_system
  30:     0x7ffbb41d01af - rustc_interface[910c478ad43be9b7]::passes::analysis
  31:     0x7ffbb3db5aaa - <dyn std[4f3d19e5cfafb1ab]::io::Write as nu_ansi_term[383177f773185f1f]::write::AnyWrite>::write_str
  32:     0x7ffbb3d2ca01 - rustc_ty_utils[8ea27dfdeb4a5d12]::ty::adt_sized_constraint
  33:     0x7ffbb3dbe577 - rustc_query_impl[f16a2bc9f261c9b3]::query_system
  34:     0x7ffbb143653d - _wpgmptr
  35:     0x7ffbb1433806 - _wpgmptr
  36:     0x7ffbb143b3a9 - _wpgmptr
  37:     0x7ffbb946832d - alloc::boxed::impl$48::call_once
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2063
  38:     0x7ffbb946832d - alloc::boxed::impl$48::call_once
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2063
  39:     0x7ffbb946832d - std::sys::pal::windows::thread::impl$0::new::thread_start
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys\pal\windows\thread.rs:52
  40:     0x7ffbfbc47374 - BaseThreadInitThunk
  41:     0x7ffbfd05cc91 - RtlUserThreadStart

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.80.0 (051478957 2024-07-21) running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
panicked at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9\compiler\rustc_hir\src\definitions.rs:81:26:
thread panicked while processing panic. aborting.
thread 'rustc' panicked at /rust/deps\indexmap-2.2.6\src\map\core.rs:37:15:
index out of bounds: the len is 3584 but the index is 9007199254744067
stack backtrace:
   0:     0x7ffbb9453b3d - std::backtrace_rs::backtrace::dbghelp64::trace
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\..\..\backtrace\src\backtrace\dbghelp64.rs:91
   1:     0x7ffbb9453b3d - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ffbb9453b3d - std::sys_common::backtrace::_print_fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:68
   3:     0x7ffbb9453b3d - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:44
   4:     0x7ffbb9484779 - core::fmt::rt::Argument::fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\fmt\rt.rs:165
   5:     0x7ffbb9484779 - core::fmt::write
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\fmt\mod.rs:1168
   6:     0x7ffbb944a391 - std::io::Write::write_fmt<std::sys::pal::windows::stdio::Stderr>
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\io\mod.rs:1835
   7:     0x7ffbb9453916 - std::sys_common::backtrace::print
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:34
   8:     0x7ffbb9456af8 - std::panicking::default_hook::closure$1
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:271
   9:     0x7ffbb94567a6 - std::panicking::default_hook
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:298
   Compiling fontdb v0.16.2
  10:     0x7ffbb550a39c - memchr
  11:     0x7ffbb9457127 - alloc::boxed::impl$50::call
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2077
  12:     0x7ffbb9457127 - std::panicking::rust_panic_with_hook
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:799
  13:     0x7ffbb9456fb7 - std::panicking::begin_panic_handler::closure$0
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:664
  14:     0x7ffbb94544af - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys_common\backtrace.rs:171
  15:     0x7ffbb9456c68 - std::panicking::begin_panic_handler
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\panicking.rs:652
  16:     0x7ffbb94ad124 - core::panicking::panic_fmt
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\panicking.rs:72
  17:     0x7ffbb94ad35e - core::panicking::panic_bounds_check
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\core\src\panicking.rs:274
  18:     0x7ffbb490dc72 - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  19:     0x7ffbb49bb699 - <rustc_resolve[8978f99e429eaae2]::Resolver>::into_outputs
  20:     0x7ffbb49b2b80 - <rustc_resolve[8978f99e429eaae2]::Resolver>::into_outputs
  21:     0x7ffbb490f6f3 - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  22:     0x7ffbb4965d57 - <rustc_resolve[8978f99e429eaae2]::late::LateResolutionVisitor as rustc_ast[552c7cfc53262d22]::visit::Visitor>::visit_fn
  23:     0x7ffbb4913e6e - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  24:     0x7ffbb4949640 - <rustc_resolve[8978f99e429eaae2]::late::LateResolutionVisitor as rustc_ast[552c7cfc53262d22]::visit::Visitor>::visit_item
  25:     0x7ffbb4913efe - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  26:     0x7ffbb494a25e - <rustc_resolve[8978f99e429eaae2]::late::LateResolutionVisitor as rustc_ast[552c7cfc53262d22]::visit::Visitor>::visit_item
  27:     0x7ffbb4913efe - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  28:     0x7ffbb494a25e - <rustc_resolve[8978f99e429eaae2]::late::LateResolutionVisitor as rustc_ast[552c7cfc53262d22]::visit::Visitor>::visit_item
  29:     0x7ffbb4913efe - <rustc_ast_lowering[a0da7e0c932a075]::LoweringContext>::lower_generic_param
  30:     0x7ffbb494a25e - <rustc_resolve[8978f99e429eaae2]::late::LateResolutionVisitor as rustc_ast[552c7cfc53262d22]::visit::Visitor>::visit_item
  31:     0x7ffbb4998a86 - <rustc_resolve[8978f99e429eaae2]::Resolver>::into_outputs
  32:     0x7ffbb3cdd7e4 - <rustc_resolve[8978f99e429eaae2]::Resolver>::resolve_crate
  33:     0x7ffbb41cdb16 - rustc_interface[910c478ad43be9b7]::passes::resolver_for_lowering_raw
  34:     0x7ffbb3db5531 - <dyn std[4f3d19e5cfafb1ab]::io::Write as nu_ansi_term[383177f773185f1f]::write::AnyWrite>::write_str
  35:     0x7ffbb3da034d - <dyn std[4f3d19e5cfafb1ab]::io::Write as nu_ansi_term[383177f773185f1f]::write::AnyWrite>::write_str
  36:     0x7ffbb3d29a14 - rustc_ty_utils[8ea27dfdeb4a5d12]::ty::adt_sized_constraint
  37:     0x7ffbb3dbce36 - rustc_query_impl[f16a2bc9f261c9b3]::query_system
  38:     0x7ffbb40ce1c0 - <rustc_middle[1c821902dd0c6a84]::ty::context::TyCtxt>::resolver_for_lowering
  39:     0x7ffbb14363d5 - _wpgmptr
  40:     0x7ffbb1433806 - _wpgmptr
  41:     0x7ffbb143b3a9 - _wpgmptr
  42:     0x7ffbb946832d - alloc::boxed::impl$48::call_once
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2063
  43:     0x7ffbb946832d - alloc::boxed::impl$48::call_once
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\alloc\src\boxed.rs:2063
  44:     0x7ffbb946832d - std::sys::pal::windows::thread::impl$0::new::thread_start
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library\std\src\sys\pal\windows\thread.rs:52
  45:     0x7ffbfbc47374 - BaseThreadInitThunk
  46:     0x7ffbfd05cc91 - RtlUserThreadStart

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.80.0 (051478957 2024-07-21) running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [resolver_for_lowering_raw] getting the resolver for lowering
end of query stack

**Comment by lqd** [comment]
Hmm, that doesn't look like the same problem. Please open a new issue with some reproducible steps, just in case. It's fine if it's a duplicate, but we'll need to make sure with the exact steps to reproduce.
